src/session/sidebar/mod.rs

```diff
@@ -201,4 +262,119 @@ impl Sidebar {
+                let chat1 = obj1.downcast_ref::<Chat>();
+                let chat2 = obj2.downcast_ref::<Chat>();
+
+                // Always show chats first and then users
+                if chat1.is_none() && chat2.is_some() {
+                    gtk::Ordering::Larger
+                } else if chat1.is_some() && chat2.is_none() {
+                    gtk::Ordering::Smaller
+                } else if chat1.is_none() && chat2.is_none() {
+                    gtk::Ordering::Equal
+                } else {
+                    let order1 = chat1.unwrap().order();
+                    let order2 = chat2.unwrap().order();
+
+                    order2.cmp(&order1).into()
+                }
+            });
+
+            session.chat_list().connect_positions_changed(
+                clone!(@weak filter, @weak sorter => move |_| {
+                    filter.changed(gtk::FilterChange::Different);
+                    sorter.changed(gtk::SorterChange::Different);
+                }),
+            );
+
+            let filter_model = gtk::FilterListModel::new(Some(&model), Some(&filter));
+            let sort_model = gtk::SortListModel::new(Some(&filter_model), Some(&sorter));
+            let selection = gtk::SingleSelection::new(Some(&sort_model));
+            selection.set_autoselect(false);
+
+            selection.connect_selected_item_notify(
+                clone!(@weak self as obj, @weak session => move |selection| {
+                    if let Some(item) = selection.selected_item() {
+                        if let Some(chat) = item.downcast_ref::<Chat>() {
+                            obj.set_selected_chat(Some(chat.to_owned()));
+                        } else if let Some(user) = item.downcast_ref::<User>() {
+                            // Create a chat with the user and then select the created chat
+                            let user_id = user.id();
+                            let client_id = session.client_id();
+                            do_async(
+                                glib::PRIORITY_DEFAULT_IDLE,
+                                async move {
+                                    functions::CreatePrivateChat::new()
+                                        .user_id(user_id)
+                                        .send(client_id)
+                                        .await
+                                },
+                                clone!(@weak obj, @weak session => move |result| async move {
+                                    if let Ok(enums::Chat::Chat(chat)) = result {
+                                        let chat = session.chat_list().get_chat(chat.id);
+                                        obj.set_selected_chat(chat);
+                                    }
+                                }),
+                            );
+                        } else {
+                            panic!("Unexpected item type: {:?}", item);
```

Comment:
If this is our responsibility to ensure that this never happens, then I would use the `unreachable!` macro.

Reply:
Was it intentional that you changed the other `panic!` but not this one? :-P If this is the case you can mark this as resolved again.

Reply:
I actually think I didn't even see that review somehow.